src/NuGet.Core/NuGet.Build.Tasks.Pack/Pack.targets

```diff
@@ -237,15 +222,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     <MSBuild
       Projects="@(ValidProjectInputForPackGraph)"
       Targets="_AddMetadataToProjectReference"
-      Properties="%(_MSBuildProjectReferenceExistent.SetConfiguration);
-              %(_MSBuildProjectReferenceExistent.SetPlatform);
-              CustomAfterMicrosoftCommonTargets=$(MSBuildThisFileFullPath);
+      Properties="CustomAfterMicrosoftCommonTargets=$(MSBuildThisFileFullPath);
               BuildProjectReferences=false;
-              TargetFramework=$(TargetFramework);
+              _ParentTargetFramework=$(TargetFramework);
```

Comment:
My suggestion to make this easier to follow would be:

1. Don't pass the parent target framework back and forth, just add the target framework in the caller after you get the other data back.

2. Rename _AddMetadataToProjectReferences to _GetPackageInfo or something and don't name the single item it returns _ProjectToProjectReferences.